plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4729,6 +4729,17 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
         return VirtualMachineGuestOsIdentifier.OTHER_GUEST;
     }
 
+    private String translateGuestOsIdentifierEx(String cpuArchitecture, String guestOs, String cloudGuestOs) {
+        if (cloudGuestOs != null) {
```

Comment:
What about inverting the logic here? So, we can reduce one if-inception.
If(cloudGuestOs == null){
return translateGuestOsIdentifier(cpuArchitecture, guestOs, cloudGuestOs).value();
}
...the rest of the code that was initially inside the if.